internal/gazelle.bash.in

```diff
@@ -17,6 +17,19 @@
 
 @@GENERATED_MESSAGE@@
 
+# Copy-pasted from the Bazel Bash runfiles library v2.
```

Comment:
Go code should use `github.com/bazelbuild/rules_go/go/tools/bazel` instead of relying on bash to access runfiles. Is there something missing in that library that this provides? I'd actually like to get rid of all bash dependencies in Gazelle, so if something is missing, I'd prefer the Go implementation to be updated.

Reply:
Not sure what's the best approach here.

The Go version fails to locate the runfiles directory because the var `RUNFILES_DIRECTORY` is not set.

Another side-effect is that I have to set `RUNFILES_MANIFEST_ONLY=1` when `exec()`ing a tool built with `rules_nodejs/nodejs_binary`, otherwise it complains about missing runfiles.

Reply:
So, I think I know what's the issue:

`github.com/bazelbuild/rules_go/go/tools/bazel` will probably try to do a very similar initialization as the bash version, but the expected environment/runtime state is different and so it fails. The bash version will be execute at the entry of gazelle runner, and so both `$0` or the current working directory (at least when executed through `bazel run`) can be used to find the runfiles directory and/or manifest.

Within the gazelle binary, we have two issues: `$0` equivalent can't be used anymore to find the correct runfiles, as that's the gazelle binary, not the runner, and the cwd has changed from inside the runfiles tree to the workspace root by gazelle runner. This way, the Go version, at least when launched by gazelle runner, doesn't work.

It seems to me that gazelle runner has to export the correct env vars for gazelle binary in any case. I don't think we need to necessarily rely on the bash library, but doing otherwise is kinda duplicating logic.

Reply:
Thanks for explaining. To summarize my understanding, it looks like Bazel doesn't set any runfiles-related environment variables when invoking a binary with `bazel run`, and since this script changes the working directory and exec's another binary, there's no way to find the runfiles.

That said, I'm not willing to take on any more dependence on Bash, since it's already causing problems for Windows users. I'd also like to centralized all Go-related handling of runfiles in `github.com/bazelbuild/rules_go/go/tools/bazel`.

I think the right approach here is to eliminate the Bash script entirely. Gazelle can run directly out of its original working directory (with `-repo_root` defaulting to `BUILD_WORKSPACE_DIRECTORY`). I'll try and make that change in the next couple days.

To provide some context, the runner script was added to work around problems with `bazel run` which have long since been fixed. I don't think it should be necessary anymore.

Reply:
Makes sense, that works for me. Let me know when changes are on master so I can rebase & test.

Reply:
#830 made this change. I wasn't able to completely remove the bash script (still need a wrapper to pass in arguments), but it no longer changes the working directory, so the Gazelle binary should be able to locate runfiles as if it were run directly with `bazel run`.

So I think if you update to `master`, then update this PR to just do the `runfiles.merge`, it should work.

Reply:
@jayconrod just tried updating to master and still no luck without `runfiles_export_envvars`:

```
gazelle: could not locate runfiles directory
```

Reply:
@greenboxal Sorry I had to revert #830 in #844 since it caused other problems. Sorry I won't have time to look at it for a while.

Reply:
Makes sense, any chance we can merge the bash-based solution?

Reply:
No, I think it would run into the same problem, and I'd still strongly prefer not to expand the dependency on bash.